MekHQ/src/mekhq/campaign/universe/PlanetarySystem.java

```diff
@@ -0,0 +1,629 @@
+    		if(null != id) {
+    			return id;
+    		}
+    	}
+        if(null != getPrimaryPlanet()) {
+            return getPrimaryPlanet().getName(when);
+        }
+        return "Unknown";     
+    }
+    
+    public List<String> getFactions(DateTime when) {
+        ArrayList<String> factions = new ArrayList<String>();
+        for(Planet planet : planets.values()) {
+            List<String> f = planet.getFactions(when);
+            if(null != f) {
+                factions.addAll(f);
+            }
+        }
+        return factions;
+    }
+    
+    public Set<Faction> getFactionSet(DateTime when) {
+        Set<Faction> factions = new HashSet<Faction>();
+        for(Planet planet : planets.values()) {
+            Set<Faction> f = planet.getFactionSet(when);
+            if(null != f) {
+                factions.addAll(f);
+            }
+        }
+        return factions;
+    }
+
+    /** highest socio-industrial ratings among all planets in system for the map **/
+    public SocioIndustrialData getSocioIndustrial(DateTime when) {
+        int tech = EquipmentType.RATING_X;
+        int industry = EquipmentType.RATING_X;
+        int rawMaterials = EquipmentType.RATING_X;
+        int output = EquipmentType.RATING_X;
+        int agriculture = EquipmentType.RATING_X;
+        
+        for(Planet planet : planets.values()) {
+            SocioIndustrialData sic = planet.getSocioIndustrial(when);
+            if(null != sic) {
+                if(sic.tech < tech) {
+                    tech = sic.tech;
+                }
+                if(sic.industry < industry) {
+                    industry = sic.industry;
+                }
+                if(sic.rawMaterials < rawMaterials) {
+                    rawMaterials = sic.rawMaterials;
+                }
+                if(sic.output < output) {
+                    output = sic.output;
+                }
+                if(sic.agriculture < agriculture) {
+                    agriculture = sic.agriculture;
+                }
+            }
+        }
+        return new SocioIndustrialData(tech, industry, rawMaterials, output, agriculture);
+    }
+    
+    /** @return the highest HPG rating among planets **/
+    public Integer getHPG(DateTime when) {
+        int rating = EquipmentType.RATING_X;
+        for(Planet planet : planets.values()) {
+            if(null != planet.getHPG(when) && planet.getHPG(when) < rating) {
+                rating = planet.getHPG(when);
+            }
+        }
+        return rating;
+    }
+    
+    /** @return short name if set, else full name, else "unnamed" */
+    public String getPrintableName(DateTime when) {
+        String result = getName(when);
+        if(null == result) {
+            return "Unknown";
+        }
+        //remove numbers or roman numerals from name
+        //not doing this anymore because connectors may have numeric endings
+        /*
+        String new_result = "";
```

Comment:
Let's figure out what to do with this commented out code.

Reply:
I removed the commented out code. The consequence is that places like "Desai IV" will show up on the map as "Desai IV" rather than "Desai" but I think the assumption that all numbers indicate planetary position is problematic with connectors and clusters.